### DISCUSSION TITLE
How to get the Active Element inside a form?

### DISCUSSION THREAD

**Discussion by Rednas83** [formkit/formkit#1462] [General]
Tried
```
import { useActiveElement } from '@vueuse/core'

const activeElement = useActiveElement()

watch(activeElement, (el) => {
  console.log('focus changed to', el.id)
})
```
But it's returning something like `input_232_3` instead of the name that corresponds to the form input.

Any idea?

**Comment by justin-schroeder** [comment]
That is probably the `id` attribute (which is autogenerated if you dont provide one)

**Comment by Rednas83** [comment]
Ok, I will add the `id` attribute for now.

Why is the name not used for autognerating the `id` when not provided?

Something like
![image](https://github.com/user-attachments/assets/56860f56-ac11-40d3-a11e-97b3af477acb)

**Comment by Rednas83** [comment]
Any thoughts? I would prefer not to add an `id` everywhere. Why not make the `autogenerated id` more readable?

**Comment by justin-schroeder** [comment]
im sure something like that could be done (using the node’s hierarchy as the id) which would certainly reduce duplicates — not eliminate them though. however even in those cases it would not be globally accessible since nodes are only put in the global registry if they are given an explicit id. This could also be done in user-land for your own project

**Comment by Rednas83** [comment]
For now my only requirement is to have a logical name of the focused element but I think there could be many more use cases for an id based on the `name`.

Something like
```
import { useActiveElement } from '@vueuse/core'

const activeElement = useActiveElement()

watch(activeElement, (el) => {
  console.log('focus changed to', el.id) // Returns form1.user.mail instead of input_232_3
})
```

Should it not be user responsibility to provide unique names? I have not tested with duplicated names but I guess there would be more serious issues then

I am not in a rush with this. Please upvote if you like to have an `autogenerated id` based on the name.
![image](https://github.com/user-attachments/assets/7d95e745-c9c9-47d2-8f42-6e97f0381004)
